### PR TITLE
triu_indices

### PR DESCRIPTION
**Pull request by Sarvesh-Kesharwani**
Closes #18033 -> new PR #18364

### CODE REVIEW

**Comment by Sarvesh-Kesharwani** [comment]
pycharm64_fIZU7WHQO1

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/frontends/paddle/tensor/creation.py [hunk 1]
```diff
@@ -114,3 +114,15 @@ def tril(x, diagonal=0, name=None):
...
+@with_supported_dtypes({"2.5.0 and below": ("int32", "int64")}, "paddle")
+@to_ivy_arrays_and_back
+def triu_indices(row, col=None, offset=0, dtype="int64"):
+    try:
+        arr = ivy.triu_indices(row, col, offset)
+        arr = ivy.astype(arr, ivy.as_native_dtype(dtype))
+        return arr
+    except Exception as e:
+        print("Generic Error: ", e)
+        return arr
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/creation.py:128]: We shouldn't be putting our code in a try block and catching the exceptions, instead let them propagate to the user.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_creation.py [hunk 2]
```diff
@@ -417,6 +417,46 @@ def test_paddle_triu(
...
+# triu_indices
+@handle_frontend_test(
+    fn_tree="paddle.triu_indices",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("integer")
+    ),
+    dtype=helpers.get_dtypes("valid", full=False),
+    row=st.integers(min_value=1, max_value=5),
+    col=st.integers(min_value=1, max_value=5),
+    offset=st.integers(min_value=0, max_value=4),
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_creation.py:436]: `offset` needs to be tested for negative values as well

File: ivy/functional/frontends/paddle/tensor/creation.py [hunk 3]
```diff
@@ -114,3 +114,15 @@ def tril(x, diagonal=0, name=None):
...
 def triu(x, diagonal=0, name=None):
     return ivy.triu(x, k=diagonal)
+
+
+@with_supported_dtypes({"2.5.0 and below": ("int32", "int64")}, "paddle")
+@to_ivy_arrays_and_back
+def triu_indices(row, col=None, offset=0, dtype="int64"):
+    try:
+        arr = ivy.triu_indices(row, col, offset)
+        arr = ivy.astype(arr, ivy.as_native_dtype(dtype))
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/creation.py:124]: No need to use the `ivy.as_native_dtype(dtype)`. Simply pass in `dtype` here. We don't work with anything native in the frontends.

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @Sarvesh-Kesharwani, requested a couple of changes in the implementation. Do ask for a re-review once done with the changes. Thanks! :)
